File: src/main/java/com/ada/conversation/infrastructure/in/rest/mapper/ChatRestMapper.java
package com.ada.conversation.infrastructure.in.rest.mapper;

import com.ada.conversation.application.dto.*; import com.ada.conversation.infrastructure.in.rest.dto.*;
import org.mapstruct.Mapper;
@Mapper(componentModel="spring") public interface ChatRestMapper { ChatRequest toApplication(ChatHttpRequest request); ChatHttpResponse toResponse(ChatResult result); default MessageStatusHttpResponse toStatus(String messageId, MessageExecutionState state) { return new MessageStatusHttpResponse(messageId, state.code(), state.detail()); } }

File: src/main/kotlin/com/ada/conversation/infrastructure/in/rest/mapper/ChatRestMapper.kt
package com.ada.conversation.infrastructure.`in`.rest.mapper

import com.ada.conversation.infrastructure.`in`.rest.dto.ChatHttpRequest
import com.ada.conversation.infrastructure.`in`.rest.dto.ChatHttpResponse
import com.ada.conversation.infrastructure.`in`.rest.dto.MessageStatusHttpResponse
import com.ada.conversation.application.dto.ChatRequest
import com.ada.conversation.application.dto.ChatResult
import com.ada.conversation.application.dto.MessageExecutionState
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(componentModel = "spring")
interface ChatRestMapper {
    fun toApplication(request: ChatHttpRequest): ChatRequest

    fun toResponse(result: ChatResult): ChatHttpResponse

    @Mapping(source = "state.code", target = "state")
    @Mapping(source = "state.detail", target = "detail")
    fun toStatus(messageId: String, state: MessageExecutionState): MessageStatusHttpResponse
}
